core/formats/columnstore.cpp

```diff
@@ -2011,8 +2011,8 @@ class sparse_column final : public column {
     return memory::make_managed<iterator_t>(
         *this,
         refs_.data(),
-        refs_.data() + refs_.size() - 1,
-        hint == ColumnHint::kNormal); // -1 for upper bound
+        refs_.data() + refs_.size() - 1, // -1 for upper bound
+        ColumnHint::kNormal == (hint & ColumnHint::kConsolidation));
```

Comment:
NIT
I don't like how it look
Maybe something like this
```suggestion
        (hint & ColumnHint::kConsolidation) != ColumnHint::kConsolidation);
```

Reply:
I mean in your code we want to know that kNormal is zero and ColumnHint values not intersect

Reply:
Done